include/simdjson/dom/serialization-inl.h

```diff
@@ -0,0 +1,361 @@
+
+#ifndef SIMDJSON_SERIALIZATION_INL_H
+#define SIMDJSON_SERIALIZATION_INL_H
+
+#include "simdjson/dom/serialization.h"
+
+#include <cinttypes>
+#include <type_traits>
+
+namespace simdjson {
+namespace dom {
+inline bool parser::print_json(std::ostream &os) const noexcept {
+  if (!valid) { return false; }
+  simdjson::dom::string_builder<> sb;
+  sb.append(doc.root());
+  std::string_view answer = sb.str();
+  os << answer;
+  return true;
+}
+/***
+ * Number utility functions
+ **/
+
+
+namespace {
+char *fast_itoa(char *output, int64_t value) noexcept {
+  // This is a standard implementation of itoa.
```

Comment:
I take it this is faster than normal itoa?

Reply:
> I take it this is faster than normal itoa?

As far as I know, itoa is not part of the standard so it may not be available some of time and it is too silly a piece of code to do complicated work to determine whether it is natively available. The modern way is "from_chars" but it is not available before C++17. I'll add a comment.

Reply:
I have documented then now. Why they are there and what they do.

Note that I am annoyed that I couldn't get the same code to work for both signed and unsigned integers, but my attempts at fusing the two either resulted in more complex code, potentially slower code, or code that triggered warnings under some compilers. In this instance, copy-paste seems acceptable.